tests/binary-with-plugin/BUILD.bazel

```diff
@@ -26,27 +28,59 @@ haskell_library(
     srcs = ["Plugin.hs"],
     deps = [
         ":base",
+        ":bytestring",
         ":ghc",
         ":process",
     ],
 )
 
 ghc_plugin(
-    name = "plugin",
-    args = ["$(location //tests/binary-simple)"],
+    name = "plugin1",
+    args = ["$(location //tests/binary-simple)", "plugin1"],
+    module = "Plugin",
+    tools = ["//tests/binary-simple"],
+    deps = [":plugin-lib"],
+)
+
+ghc_plugin(
+    name = "plugin2",
+    args = ["$(location //tests/binary-simple)", "plugin2"],
     module = "Plugin",
     tools = ["//tests/binary-simple"],
     deps = [":plugin-lib"],
 )
 
 haskell_test(
-    name = "binary-with-plugin",
+    name = "binary-with-plugin1",
     srcs = ["Main.hs"],
     plugins = select({
         # XXX If profiling is enabled, ignore the plugin because of
         # https://gitlab.haskell.org/ghc/ghc/issues/14335.
         ":debug_build": [],
-        "//conditions:default": [":plugin"],
+        "//conditions:default": [":plugin1"],
+    }),
+    visibility = ["//visibility:public"],
+    deps = [":base"],
+)
+
+haskell_test(
+    name = "binary-with-plugin2",
+    srcs = ["main2.hs"],
+    compiler_flags = ["-fplugin=Plugin"],
+    non_default_plugins = select({
+        ":debug_build": [],
+        "//conditions:default": [":plugin2"],
+    }),
+    visibility = ["//visibility:public"],
+    deps = [":base"],
+)
```

Comment:
I think these tests fail in CI because the main program is now checking if the plugin is enabled or not. Is there some way to skip these tests in the profiling configuration?

Reply:
Yes, you can add `tags = ["skip_profiling"]`.